staging/src/k8s.io/apiserver/pkg/endpoints/watch_test.go

```diff
@@ -212,6 +212,47 @@ func TestWatchWebsocketClientClose(t *testing.T) {
 	}
 }
 
+func TestWatchClientClose(t *testing.T) {
+	simpleStorage := &SimpleRESTStorage{}
+	_ = rest.Watcher(simpleStorage) // Give compile error if this doesn't work.
+	handler := handle(map[string]rest.Storage{"simples": simpleStorage})
+	server := httptest.NewServer(handler)
+	defer server.Close()
+
+	dest, _ := url.Parse(server.URL)
+	dest.Path = "/" + prefix + "/" + testGroupVersion.Group + "/" + testGroupVersion.Version + "/simples"
+	dest.RawQuery = "watch=1"
+
+	request, err := http.NewRequest("GET", dest.String(), nil)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	request.Header.Add("Accept", "application/json")
+
+	response, err := http.DefaultClient.Do(request)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if response.StatusCode != http.StatusOK {
+		b, _ := ioutil.ReadAll(response.Body)
+		t.Fatalf("Unexpected response: %#v\n%s", response, string(b))
+	}
+
+	if err := response.Body.Close(); err != nil {
```

Comment:
perhaps add a comment here saying `// Close response without reading it to cause a cancel on the server`

Reply:
I have added comment, thanks